Adding ETDump integration to coreml_executor_runner

tarun292 opened the pull request:
Differential Revision: D51990610

```bash
# Should see mv3_coreml_etrecord.bin generated after this command.
python3 -m examples.apple.coreml.scripts.export_and_delegate --model_name mv3 --generate_etrecord
# Build coreml runner
sh examples/apple/coreml/scripts/build_executor_runner.sh
# Should see etdump.etdp generated in this folder after running this command.
./coreml_executor_runner --model_path mv3_coreml_all.pte
# Run the inspector cli tool to print the profiling results
python3 -m sdk.inspector.inspector_cli --etdump_path etdump.etdp --etrecord_path mv3_coreml.bin 
╒════╤════════════════════╤══════════════════════╤════════════╤════════════╤════════════╤════════════╤════════════╤════════════╤════════════╤═══════════════════╤═════════════════════════╕
│    │ event_block_name   │ event_name           │   p10 (ms) │   p50 (ms) │   p90 (ms) │   avg (ms) │   min (ms) │   max (ms) │ op_types   │ is_delegated_op   │ delegate_backend_name   │
╞════╪════════════════════╪══════════════════════╪════════════╪════════════╪════════════╪════════════╪════════════╪════════════╪════════════╪═══════════════════╪═════════════════════════╡
│  0 │ Default            │ Method::init         │  87.4569   │  87.4569   │  87.4569   │  87.4569   │  87.4569   │  87.4569   │ []         │ False             │                         │
├────┼────────────────────┼──────────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼───────────────────┼─────────────────────────┤
│  1 │ Default            │ Program::load_method │  87.4654   │  87.4654   │  87.4654   │  87.4654   │  87.4654   │  87.4654   │ []         │ False             │                         │
├────┼────────────────────┼──────────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼───────────────────┼─────────────────────────┤
│  2 │ Execute            │ DELEGATE_CALL        │   0.893667 │   0.893667 │   0.893667 │   0.893667 │   0.893667 │   0.893667 │ []         │ False             │                         │
├────┼────────────────────┼──────────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼───────────────────┼─────────────────────────┤
│  3 │ Execute            │ Method::execute      │   0.895833 │   0.895833 │   0.895833 │   0.895833 │   0.895833 │   0.895833 │ []         │ False             │                         │
╘════╧════════════════════╧══════════════════════╧════════════╧════════════╧════════════╧════════════╧════════════╧════════════╧════════════╧═══════════════════╧═════════════════════════╛
```

cymbalrush reviewed (approved):
LGTM, thanks Tarun!

tarun292 committed:
Adding ETDump integration to coreml_executor_runner (#1382)

Summary:
```bash
# Should see mv3_coreml_etrecord.bin generated after this command.
python3 -m examples.apple.coreml.scripts.export_and_delegate --model_name mv3 --generate_etrecord
# Build coreml runner
sh examples/apple/coreml/scripts/build_executor_runner.sh
# Should see etdump.etdp generated in this folder after running this command.
./coreml_executor_runner --model_path mv3_coreml_all.pte
# Run the inspector cli tool to print the profiling results
python3 -m sdk.inspector.inspector_cli --etdump_path etdump.etdp --etrecord_path mv3_coreml.bin 
╒════╤════════════════════╤══════════════════════╤════════════╤════════════╤════════════╤════════════╤════════════╤════════════╤════════════╤═══════════════════╤═════════════════════════╕
│    │ event_block_name   │ event_name           │   p10 (ms) │   p50 (ms) │   p90 (ms) │   avg (ms) │   min (ms) │   max (ms) │ op_types   │ is_delegated_op   │ delegate_backend_name   │
╞════╪════════════════════╪══════════════════════╪════════════╪════════════╪════════════╪════════════╪════════════╪════════════╪════════════╪═══════════════════╪═════════════════════════╡
│  0 │ Default            │ Method::init         │  87.4569   │  87.4569   │  87.4569   │  87.4569   │  87.4569   │  87.4569   │ []         │ False             │                         │
├────┼────────────────────┼──────────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼───────────────────┼─────────────────────────┤
│  1 │ Default            │ Program::load_method │  87.4654   │  87.4654   │  87.4654   │  87.4654   │  87.4654   │  87.4654   │ []         │ False             │                         │
├────┼────────────────────┼──────────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼───────────────────┼─────────────────────────┤
│  2 │ Execute            │ DELEGATE_CALL        │   0.893667 │   0.893667 │   0.893667 │   0.893667 │   0.893667 │   0.893667 │ []         │ False             │                         │
├────┼────────────────────┼──────────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼───────────────────┼─────────────────────────┤
│  3 │ Execute            │ Method::execute      │   0.895833 │   0.895833 │   0.895833 │   0.895833 │   0.895833 │   0.895833 │ []         │ False             │                         │
╘════╧════════════════════╧══════════════════════╧════════════╧════════════╧════════════╧════════════╧════════════╧════════════╧════════════╧═══════════════════╧═════════════════════════╛
```


Reviewed By: cccclai

Differential Revision: D51990610

Pulled By: tarun292
